### PR TITLE
simplify use of sockaddr* structs to work around buffer overflow warning

### PR DESCRIPTION
**Pull request by JohnWestlund**
The current method of moving around the sockaddr data causes compile time buffer overflow warnings:
> [  197s] /usr/include/bits/string3.h:84:3: warning: call to __builtin___memset_chk will always overflow destination buffer [enabled by default]
> [  197s]    return __builtin___memset_chk (__dest, __ch, __len, __bos0 (__dest));

by doing the following:
create a temp `sockaddr` struct pointer that is used in calling parse_uri
`memcpy` 0's over the struct using `sizeof `a `sockaddr_in` or `sockaddr_in6` (which is larger and causes the buffer overflow warning)
then upon return
create a `mca_oob_tcp_addr_t `object
`memcpy` the original `sockaddr` struct over `mca_oob_tcp_addr_t->addr` (which is a `sockaddr_storage` struct)

Instead this patch simplifies the process by creating the `mca_oob_tcp_addr` object first, then passes the pointer to its `sockaddr_storage` struct so the `parse_uri` copies in the data directly.

### CODE REVIEW

File: orte/mca/oob/tcp/oob_tcp.c [hunk 1]
```diff
@@ -301,8 +301,7 @@ static void process_set_peer(int fd, short args, void *cbdata)
                         ORTE_NAME_PRINT(&pop->peer),
                         (NULL == pop->net) ? "NULL" : pop->net,
                         (NULL == pop->port) ? "NULL" : pop->port);
-    maddr = OBJ_NEW(mca_oob_tcp_addr_t);
-    memcpy(&maddr->addr, &inaddr, sizeof(inaddr));
+    /* memcpy(&maddr->addr, &inaddr, sizeof(inaddr)); */
```
Comment (rhc54) [inline orte/mca/oob/tcp/oob_tcp.c]: Why leave this line commented out here? If it needs to be deleted, then let's delete it.

File: orte/mca/oob/tcp/oob_tcp.c [hunk 2]
```diff
@@ -290,7 +289,8 @@ static void process_set_peer(int fd, short args, void *cbdata)
         }
     }
 
-    if ((rc = parse_uri(pop->af_family, pop->net, pop->port, (struct sockaddr*) &inaddr)) != ORTE_SUCCESS) {
+    maddr = OBJ_NEW(mca_oob_tcp_addr_t);
+    if ((rc = parse_uri(pop->af_family, pop->net, pop->port, (struct sockaddr_storage*) &(maddr->addr))) != ORTE_SUCCESS) {
```
Comment (rhc54) [inline orte/mca/oob/tcp/oob_tcp.c]: If you get the error case, where do you release the maddr memory? Also, please put the constant first in the != check.

**Comment by hjelmn** [comment]
Is this related to the error coverity is reporting? See CID 710596

**Comment by JohnWestlund** [comment]
@hjelmn That looks like an OPAL defect, this is in ORTE.

**Comment by rhc54** [comment]
@hjelmn It's a similar problem, but we are only addressing it in the OOB - someone else would have to look at the problem in the TCP BTL.

**Comment by rhc54** [comment]
@JohnWestlund once this clears the automated checks, I think it is good to go.

**Comment by JohnWestlund** [comment]
@rhc54 Great!

**Pull request merged**